frontends/web/src/hooks/keyboard.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect } from 'react'
+
+/**
+ * useEsc handles ESC key.
+ * gets fired on ESC keydown and exectues the provided callback.
+ */
+export const useEsc = (
+    callback: () => void
+) => {
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (e.key === 'Escape') {
+                callback();
+            }
+        }
+        document.addEventListener('keydown', handleKeyDown);
+        return () => document.removeEventListener('keydown', handleKeyDown);
+    }, [callback]);
```

Comment:
Is the `[callback]` dep needed? According to the React docs:

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. 

And this effect is sth. you want to run and cleanup once on mount and unmount.

Reply:
the dependency list just tells the useEffect hook to execute (cleanup/unmount and mount again) when the array changes, in this case when the callback changes. 

Unfortunately functions cannot be compared unless it is two references to the same function. I was thinking of caching the function with useCallback but does not really improve anything afaik here and makes it harder to read.

Reply:
But how can the callback change? It is a function param.

Reply:
@benma please take a look at this codepen, clicking or pressing ESC counts up and in the codepen's console you should see mount and unmount logs. Now remove `callback` from the dependencyList and test with pressing ESC, you'll see it will only count to 1, because the useEffect in the hook isn't updated after the parent re-rendered.

 https://codepen.io/drugtgkddngme/pen/eYyBgeY?editors=1011

![Screen Shot 2022-03-23 at 8 27 39 AM](https://user-images.githubusercontent.com/546900/159645994-0d9eaf70-8c5a-4d42-9e76-fb55aac98019.png)

Reply:
nice explanation, thanks! ack